### PR TITLE
[Minor] Fix #142, check softmax fallback paths

### LINKED ISSUE #142: Follow up on #141 with a proper unit test
# 🐛 Bug
Make sure that #141 cannot happen anymore, add a small unit test to guard parity

### PR DESCRIPTION
**Pull request by blefaudeux**
## What does this PR do?
Fixes #142, check that the triton softmax fallbacks are correct.

## Before submitting

- [ ] Did you have fun?
  - Make sure you had fun coding 🙃
- [ ] Did you read the [contributor guideline](https://github.com/facebookresearch/fairscale/blob/master/CONTRIBUTING.md)?
- [ ] Was this discussed/approved via a Github issue? (no need for typos, doc improvements)
  - [ ] N/A
- [ ] Did you make sure to update the docs?
  - [ ] N/A
- [ ] Did you write any new necessary tests?
  - [ ] N/A
- [ ] Did you update the [changelog](https://github.com/facebookresearch/fairscale/blob/master/CHANGELOG.md)? (if needed)
  - [ ] N/A

## PR review
Anyone in the community is free to review the PR once the tests have passed.
If we didn't discuss your PR in Github issues there's a high chance it will not be merged.

### CODE REVIEW

**Comment by blefaudeux** [comment]
Adding a safety net so that we don't break this again, increasing the code coverage as a nice side effect. #superminor

**Comment by blefaudeux** [comment]
assuming this is ok..

File: xformers/triton/softmax.py [hunk 1]
```diff
@@ -200,7 +200,7 @@ def _softmax_dispatch(
         _triton_registered_warnings = True
 
     if mask is not None:
-        x += mask
+        x = x + mask
```
Comment (dianaml0) [inline xformers/triton/softmax.py:203]: @blefaudeux how come this change?

File: tests/test_triton_softmax.py [hunk 2]
```diff
@@ -103,3 +103,61 @@ def test_softmax_fp16(dtype):
...
+    )
+    y_triton = (
+        triton_log_softmax(X_, mask, causal)
+        if log
+        else triton_softmax(X_, mask, causal)
+    )
+
+    assert torch.allclose(y_torch, y_triton, equal_nan=True)
+
+    print(y_torch)
```
Comment (dianaml0) [inline tests/test_triton_softmax.py:149]: Awesome to have all this testing! Not a bit deal but should we remove some of the print statements here?

Comment (blefaudeux) [reply tests/test_triton_softmax.py:149]: oh yes sure, sorry about that !

File: xformers/triton/softmax.py (hunk 1 above)
Comment (blefaudeux) [reply xformers/triton/softmax.py:203]: it changes the operation from inplace to not-in place, looks like nothing but this was breaking the backward pass (and was not tested before, so this flew under the radar)

**Pull request merged**